src/esp/bindings_js/modules/simenv_embind.js

```diff
@@ -31,8 +35,10 @@ class SimEnv {
    */
   reset() {
     this.sim.reset();
-    const agent = this.sim.getAgent(this.selectedAgentId);
-    agent.setState(this.initialAgentState, true);
+    if (this.initialAgentState !== null) {
```

Comment:
Why not:

`if (this.initialAgentState) {
`

Seems like it would be more robust and simpler.

Reply:
I am generally wary of comparing by using truthy value of the object itself. It can mean a lot of things in JavaScript. That's why I like to be specific to what I am compare to.